File: prompts/fixtures/js/debug_sleep.js
"use strict";

let counter = 0;                       // line 3

function tick() {
    counter++;                         // line 6
    console.log(`tick ${counter}`);    // line 7
}

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));  // line 11
}

async function main() {
    console.log(`pid: ${process.pid}`);    // line 15
    for (let i = 0; i < 300; i++) {        // line 16
        tick();                             // line 17
        await sleep(1000);                 // line 18
    }
}

main();

File: prompts/fixtures/rust/debug_sleep.rs
use std::thread;
use std::time::Duration;

static mut COUNTER: i32 = 0;          // line 4

fn tick() {
    unsafe {
        COUNTER += 1;                  // line 8
        println!("tick {}", COUNTER);  // line 9
    }
}

fn main() {
    println!("pid: {}", std::process::id());  // line 14
    for _ in 0..300 {                          // line 15
        tick();                                // line 16
        thread::sleep(Duration::from_secs(1)); // line 17
    }
}
